hazelcast/src/main/java/com/hazelcast/internal/diagnostics/Diagnostics.java

```diff
@@ -115,6 +117,8 @@
     private final boolean enabled;
     private ScheduledExecutorService scheduler;
     private final HazelcastThreadGroup hzThreadGroup;
+    private final ConcurrentMap<Class<? extends DiagnosticsPlugin>, DiagnosticsPlugin> pluginsMap
```

Comment:
I know. 

I'm still not sure how the final design should look like for latency tracking. Currently we have a cheap way to track some counters, so I'm fine with having them enabled even if diagnostics is disabled. But I'm not looking forward to sprinkle the code with latency measuring code even though it perhaps isn't used.

So I'm not yet sure how the metrics api should be changed. Another issue is that the metrics system, dumps all metrics. But in this case, if we would make use of metrics, I don't want the metrics writer do dump all metrics; some metrics have their own renderer (in this case StoreLatencyPlugin). 

So I understand that the current design is not optimal. But I don't have a clear vision how it should look like before embarking on a big metrics change. Eventually this latency tracking stuff will be merged in metrics..  but currently I don't see a clear way how to do it. 

Another issue is that I have 5 days for this task; adding this stuff into metrics was not part of that estimation.